## 1.1.3 Bài tập thực hành 1
### Thực hiện thống kê mô tả trên tập dữ liệu về phân loại chất lượng rượu đỏ.

Giải thích ý nghĩa các chỉ số

**Mean** (trung bình cộng): cho biết mức chất lượng trung bình của rượu.

**Median** (trung vị): mức chất lượng nằm ở giữa khi sắp xếp toàn bộ dữ liệu.

**Mode** (mode): mức chất lượng phổ biến nhất.

**Variance** & **Standard Deviation**: cho thấy dữ liệu phân tán nhiều hay ít.

**Min** – **Max**: mức chất lượng thấp nhất và cao nhất.

**Percentile 60%**: tại vị trí 60% dữ liệu.

**Quartile** & **IQR**: dùng để phân tích sự phân tán và phát hiện ngoại lệ.

In [1]:
import numpy as np
import pandas as pd
from scipy import stats

# Load the .csv into a dataframe using read_csv
wine_data = pd.read_csv("winequality-red.csv")
wine_data = wine_data[['fixed acidity','volatile acidity','citric acid','alcohol','quality']]

# Take a quick look at the data
wine_data.head(5)
wine_data.dtypes
wine_data.shape

# Get the mean of the data (trung bình cộng)
data_mean = np.mean(wine_data["quality"])
# Get the median of the data (trung vị)
data_median = np.median(wine_data["quality"])
# Get the mode of the data (giá trị xuất hiện nhiều nhất)
data_mode = stats.mode(wine_data["quality"], keepdims=True)
# Obtain the variance of the data (phương sai)
data_variance = np.var(wine_data["quality"])
# Obtain the standard deviation of the data (độ lệch chuẩn)
data_sd = np.std(wine_data["quality"])
# Compute the maximum and minimum values of the data (giá trị lớn - nhỏ nhất)
data_max = np.max(wine_data["quality"])
data_min = np.min(wine_data["quality"])
# Obtain the 60th percentile of the data (phân vị, ví dụ 60%)
data_percentile = np.percentile(wine_data["quality"], 60)
# Obtain the quartiles of the data (phân vị thứ 3 - Q3)
data_quartile = np.quantile(wine_data["quality"], 0.75)
# Get the IQR of the data (khoảng tứ phân vị: Q3 - Q1)
data_IQR = stats.iqr(wine_data["quality"])
#Result
print("Mean:", data_mean)
print("Median:", data_median)
print("Mode:", data_mode.mode[0])
print("Variance:", data_variance)
print("Standard Deviation:", data_sd)
print("Min:", data_min)
print("Max:", data_max)
print("60th Percentile:", data_percentile)
print("75th Quantile (Q3):", data_quartile)
print("IQR:", data_IQR)

Mean: 5.6360225140712945
Median: 6.0
Mode: 5
Variance: 0.6517605398308277
Standard Deviation: 0.8073168769639513
Min: 3
Max: 8
60th Percentile: 6.0
75th Quantile (Q3): 6.0
IQR: 1.0


#### Nhận xét:
Giá trị trung bình (Mean) của chất lượng rượu đỏ là khoảng 5.64, gần sát với trung vị (Median = 6.0). Điều này cho thấy phân phối dữ liệu tương đối cân đối, không bị lệch nhiều.

Mode = 5 cho biết điểm chất lượng xuất hiện nhiều nhất trong tập dữ liệu là 5.

Khoảng dao động của chất lượng nằm trong khoảng từ 3 đến 8, tức là không có rượu chất lượng quá thấp (1–2) hoặc quá cao (9–10).

Độ lệch chuẩn ≈ 0.81 khá nhỏ, chứng tỏ chất lượng rượu tập trung quanh giá trị trung bình, ít phân tán.

Phương sai ≈ 0.65 phù hợp với nhận định trên (ít biến động).

Phân vị 60% và 75% đều bằng 6, cho thấy phần lớn rượu có chất lượng từ trung bình khá trở lên.

IQR = 1.0 → 50% dữ liệu tập trung trong khoảng hẹp (từ Q1 đến Q3), nghĩa là độ chênh lệch giữa nhóm rượu trung bình và khá không lớn.

Tổng thể, dữ liệu cho thấy chất lượng rượu đỏ chủ yếu nằm ở mức trung bình (5–6 điểm), chỉ một phần nhỏ đạt mức cao (7–8), rất ít trường hợp rơi vào mức thấp (3–4)